### DISCUSSION TITLE
[Laravel + Octane] FrankenPHP must be in worker mode to use this script.

### DISCUSSION THREAD

**Discussion by gnmpwr** [php/frankenphp#1189] [Q&A]
Hi there! When I try to run the Laravel application with Octane in standalone binary, I get a warning "FrankenPHP must be in worker mode to use this script.".  I did everything according to the instructions on this page: https://frankenphp.dev/docs/laravel /. Yes, I could run the project in Docker or some other way, as I did before, but this is a matter of principle, I want to figure it out.

**System information and playback steps:**
1. I use WSL2 on Windows 10. Ubuntu 20.04 / Frankenphp 1.3.1 (also try 1.2.5)
2. Install fresh Laravel (11.33.2) from this page: https://laravel.com/docs/11.x/installation#docker-installation-using-sail . Yes, im using Laravel Sail
3. Install Octane etc. 
4. Set Dockerfile  (im describe it below) 
5. Build by guide page and extract binary
6. Start server by command: PATH="$PWD:$PATH" frankenphp php-cli artisan octane:frankenphp (options if any) 
7. Got the warning "FrankenPHP must be in worker mode to use this script.". In same time server is works and send this messages with status code 200. 

**Current Dockerfile:** 

```php
FROM dunglas/frankenphp:static-builder

# Copy your app
WORKDIR /go/src/app/dist/app
COPY . .

# Remove the tests and other unneeded files to save space
# Alternatively, add these files to a .dockerignore file
RUN rm -Rf tests/

# Copy .env file
RUN cp .env.example .env
# Change APP_ENV and APP_DEBUG to be production ready
RUN sed -i'' -e 's/^APP_ENV=.*/APP_ENV=production/' -e 's/^APP_DEBUG=.*/APP_DEBUG=false/' -e 's/^OCTANE_SERVER=.*/OCTANE_SERVER=frankenphp/' .env

# Make other changes to your .env file if needed
ARG MIMALLOC
ARG PHP_VERSION=8.4
ARG PHP_EXTENSIONS="bcmath calendar curl dom fileinfo gd intl mbstring mysqli opcache pdo pdo_mysql xml zip"
# Install the dependencies
RUN composer install --ignore-platform-reqs --no-dev -a
RUN composer dump-autoload

ARG CLEAN
# Build the static binary
WORKDIR /go/src/app/
RUN rm -rf dist/static-php-cli/watcher
ENV NO_COMPRESS=1
RUN EMBED=dist/app/ ./build-static.sh
```

I must say right away: I have tried different versions of Dockerfile. The standard one, which is specified in the instructions, various other options. Also im try to run server with commands like: FRANKENPHP_MODE=worker PATH="$PWD:$PATH" ./frankenphp php-cli artisan octane:frankenphp --host=0.0.0.0 --port=8000 and other variations. 

Right now i want try fresh install without Laravel Sail. 

Thank you in advance for your help, ideas and suggestions. If you need additional information, let me know

**Comment by AlliBalliBaba** [comment]
That is indeed weird, are you using your own Caddyfile or just the one built in by Octane? I'll try if I can reproduce your setup.

**Reply by gnmpwr** [reply]
> That is indeed weird, are you using your own Caddyfile or just the one built in by Octane? I'll try if I can reproduce your setup.

Default settings and files. No changes. Im try to change Caddy file right now but no results =)

**Reply by AlliBalliBaba** [reply]
I think the issue has something to do with directories. The temporary directory from the app somehow seems to get lost along the way. It's also possible to run octane without `artisan octane:start though`. I haven't gotten around to building my own binary yet, but here is something you could try:

1. Add a `Caddyfile` to the root of your repository. Here's a default configuration working with Laravel Octane:

```
{
	admin localhost:2019

	frankenphp {
		worker {
		    file "public/frankenphp-worker.php"
		    watch
		}
	}
}

:8000 {
	log {
		level warn
	}

	route {
		root public
		encode zstd br gzip

		php_server {
			index frankenphp-worker.php
			resolve_root_symlink false
		}
	}
}

```

2. Modify the content of the file in` public/frankenphp-worker.php` (you can also commit or adapt these changes):

```
<?php

$_ENV['APP_ENV'] = 'production';
$_ENV['APP_BASE_PATH'] = __DIR__ . '/..';
$_ENV['APP_PUBLIC_PATH'] = __DIR__;
$_ENV['LARAVEL_OCTANE'] = 1;
$_ENV['MAX_REQUESTS'] = 1000;
$_ENV['REQUEST_MAX_EXECUTION_TIME'] = 30;

require __DIR__ . '/../vendor/laravel/octane/bin/frankenphp-worker.php';
```

Step 3: build the binary with the embedded app

Step 4: start the app with `frankenphp php-server`. Alternatively you can also start the app with `frankenphp run -c Caddyfile` (I think), in that case the `Caddyfile` doesn't need to be embedded, but can just be in the same directory as the binary.

**Reply by gnmpwr** [reply]
Hmm, its works. Both variants in binary, like frankenphp php-server and frankenphp php-cli artisan octane:frankenphp. But second variant have 10-12 sec requests )) Anyway big thanks! This brought some clarity to the problem. I'll try to finish it.

**Reply by AlliBalliBaba** [reply]
Yeah the `artisan octane:frankenphp` command will create a 'wrapper' process around FrankenPHP. In some cases this process is unnecessary and can badly impact performance. The main idea is to be able to stop and start FrankenPHP with artisan commands. I'm planning on making a PR to Octane that allows also running it without the wrapper, I'm just not sure yet how to best inject the configuration in this case
